### PR TITLE
fix(docker): unable to build image using podman

### PR DESCRIPTION
**Pull request by Omar-AlJaljuli**
## Description

Created a script to update the Dockerfile with every build to 'COPY' package.json files without using --parents to allow use of podman.

## Which issue(s) does this PR fix

https://issues.redhat.com/browse/RHIDP-4817

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -53,7 +53,8 @@ RUN chmod +x "$YARN"
 FROM skeleton AS deps
 COPY $EXTERNAL_SOURCE_NESTED/dynamic-plugins/ ./dynamic-plugins/
 COPY $EXTERNAL_SOURCE_NESTED/package.json $EXTERNAL_SOURCE_NESTED/yarn.lock ./
```
Comment (nickboldt) [inline docker/Dockerfile:55]:
```suggestion
COPY $EXTERNAL_SOURCE_NESTED/yarn.lock ./
```

File: scripts/update-Dockerfile.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+for dockerfile in ./docker/Dockerfile .rhdh/docker/Dockerfile; do
+  sed -i "/# BEGIN COPY package.json files/,/# END COPY package.json files/c# BEGIN COPY package.json files\n# END COPY package.json files" $dockerfile
+  for path in $(find . -name package.json | grep -v node_modules/ ); do
```
Comment (nickboldt) [inline scripts/update-Dockerfile.sh:4]:
```suggestion
  for path in $(find . -name package.json | grep -v node_modules/ | sort -uV); do
```

**Review by nickboldt — APPROVED** [review]

File: scripts/update-Dockerfile.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+for dockerfile in ./docker/Dockerfile .rhdh/docker/Dockerfile; do
+  sed -i "/# BEGIN COPY package.json files/,/# END COPY package.json files/c# BEGIN COPY package.json files\n# END COPY package.json files" $dockerfile
+  for path in $(find . -name package.json | grep -v node_modules/ | sort -uV); do
```
Comment (nickboldt) [inline scripts/update-Dockerfile.sh:4]:
```suggestion
  for path in $(find . -name package.json | grep -E -v "dynamic-plugins/|node_modules/" | sort -uV); do
```

File: .rhdh/docker/Dockerfile [hunk 4]
```diff
@@ -51,17 +51,18 @@ RUN chmod +x "$YARN"
 
 # Stage 2 - Install dependencies
 COPY $EXTERNAL_SOURCE_NESTED/dynamic-plugins/ ./dynamic-plugins/
```
Comment (nickboldt) [inline .rhdh/docker/Dockerfile:53]: I think @schultzp2020 would prefer if we don't do this line and instead DO include the dynamic-plugins/ folders in update-Dockerfile.sh line 4

Comment (Omar-AlJaljuli) [reply .rhdh/docker/Dockerfile:53]: Do we remove `$EXTERNAL_SOURCE_NESTED/dynamic-plugins/ ./dynamic-plugins/` entirely, or does he prefer that we keep it and still copy it's package.json files?

File: scripts/update-Dockerfile.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+for dockerfile in ./docker/Dockerfile .rhdh/docker/Dockerfile; do
+  sed -i "/# BEGIN COPY package.json files/,/# END COPY package.json files/c# BEGIN COPY package.json files\n# END COPY package.json files" $dockerfile
+  for path in $(find . -name package.json | grep -E -v "node_modules/" | sort -uV); do
+    sed -i "s|\# BEGIN COPY package.json files|\# BEGIN COPY package.json files\nCOPY ${path/\./\$EXTERNAL_SOURCE_NESTED} $path|g" $dockerfile
+  done
+done
```
Comment (schultzp2020) [inline scripts/update-Dockerfile.sh:7]:
```suggestion
done

```

**Review by schultzp2020 — DISMISSED** [review]

**Pull request merged**